### DISCUSSION TITLE
Can't create optionally authenticated path in FastAPI

### DISCUSSION THREAD

**Discussion by kschniedergers** [fief-dev/fief#150] [Q&A] [answered]
I'm trying to create an api route where you can be authenticated, but you don't have to. According to the docs, I can just add "optional=True" inside the authenticated call inside the Depends(), however I'm still getting "401 Unauthorized" and none of my code is being run. The function also works if i comment out the access_token_info stuff. Code below:
`@tournament_router.get("/tournament/{tournament_id}")
async def get_tournament(
    tournament_id: PydanticObjectId,
    access_token_info: Optional[FiefAccessTokenInfo] = Depends(
        fief_auth.authenticated(optional=True)),
):
    pass`

Sorry if I'm missing something simple

**Comment by fief-bailiff** [comment]
Hail, @kschniedergers 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
Hey @kschniedergers 👋 

It looks like you're using FastAPI. Did you make sure to set `auto_error=False` to the `scheme`:

```py
scheme = OAuth2AuthorizationCodeBearer(  
    "https://example.fief.dev/authorize",  
    "https://example.fief.dev/api/token",  
    scopes={"openid": "openid", "offline_access": "offline_access"},
    auto_error=False,  # 👈 That's important 
)
```

**Reply by kschniedergers** [reply]
That was it! Thanks :)
